pkg/order/client_test.go

```diff
@@ -298,3 +298,684 @@ func TestCreate(t *testing.T) {
+			fields: fields{
+				cfg: &config.Config{
+					Requester: &httpclient.Mock{
+						DoMock: func(req *http.Request) (*http.Response, error) {
+							body := `{
+										"payment_method": {
+											"type": "credit_card",
+											"installments": 2,
+											"statement_descriptor": "updated statement"
+										}
+									}`
+							return &http.Response{
+								StatusCode: http.StatusOK,
+								Body:       io.NopCloser(strings.NewReader(body)),
+								Header:     make(http.Header),
+							}, nil
+						},
+					},
+				},
+			},
+			args: args{
+				ctx:           context.Background(),
+				orderID:       "Order123",
+				transactionID: "Pay_Order123",
+				request: PaymentRequest{
+
+					PaymentMethod: &PaymentMethodRequest{
+						Type:                "credit_card",
+						Installments:        2,
+						StatementDescriptor: "updated statement",
+					},
+				},
+			},
+			want: &PaymentResponse{
+				PaymentMethod: &PaymentMethodResponse{
+					Type:                "credit_card",
+					Installments:        2,
+					StatementDescriptor: "updated statement",
+				},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &client{
+				cfg: tt.fields.cfg,
+			}
+			got, err := c.UpdateTransaction(tt.args.ctx, tt.args.orderID, tt.args.transactionID, tt.args.request)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Update Transaction() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("Update Transaction() got = %v, want = %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestCancelOrder(t *testing.T) {
+	type fields struct {
+		cfg *config.Config
+	}
+	type args struct {
+		ctx     context.Context
+		orderID string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    *Response
+		wantErr bool
+	}{
+		{
+			name: "should_fail_to_cancel_order",
+			fields: fields{
+				cfg: &config.Config{
+					Requester: &httpclient.Mock{
+						DoMock: func(req *http.Request) (*http.Response, error) {
+							return nil, fmt.Errorf("error canceling order")
+						},
+					},
+				},
+			},
+			args: args{
+				ctx:     context.Background(),
+				orderID: "invalidOrderID",
+			},
+			want:    nil,
+			wantErr: true,
+		},
+		{
+			name: "should_succeed_to_cancel_order",
+			fields: fields{
+				cfg: &config.Config{
+					Requester: &httpclient.Mock{
+						DoMock: func(req *http.Request) (*http.Response, error) {
+							body := `{"id": "validOrderID", "status": "cancelled"}`
+							return &http.Response{
+								StatusCode: http.StatusOK,
+								Body:       io.NopCloser(strings.NewReader(body)),
+								Header:     make(http.Header),
+							}, nil
+						},
+					},
+				},
+			},
+			args: args{
+				ctx:     context.Background(),
+				orderID: "validOrderID",
+			},
+			want: &Response{
+				ID:     "validOrderID",
+				Status: "cancelled",
```

Comment:
[nth] Aqui seria cancelled com um L só, não?
```suggestion
				Status: "canceled",
```